### PR TITLE
Workflow refactor

### PR DESCRIPTION
**Pull request by beaugogh**

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -102,27 +102,6 @@
           }
         }
       }
-    },
-    "glowing-bear-e2e": {
```
Comment (beaugogh) [inline angular.json]: since we are using cypress, this config is no longer needed

File: package.json [hunk 2]
```diff
@@ -9,25 +9,25 @@
...
+    "@angular/common": "^7.0.1",
+    "@angular/compiler": "^7.0.1",
+    "@angular/compiler-cli": "^7.0.1",
+    "@angular/core": "^7.0.1",
+    "@angular/forms": "^7.0.1",
+    "@angular/http": "^7.0.1",
+    "@angular/platform-browser": "^7.0.1",
+    "@angular/platform-browser-dynamic": "^7.0.1",
+    "@angular/platform-server": "^7.0.1",
+    "@angular/router": "^7.0.1",
```
Comment (beaugogh) [inline package.json]: this upgrade fixes ng6's regression on lazy loading issues

File: src/styles/styles.css [hunk 3]
```diff
@@ -253,327 +253,3 @@ body .hline {
 .form-control:-ms-input-placeholder {
   color: lightgray;
 }
-
```
Comment (beaugogh) [inline src/styles/styles.css]: moving component-wise styles to their respective component css, keep only the global css in styles.css

File: src/tests/cohort.integration.spec.ts [hunk 4]
```diff
@@ -19,30 +19,15 @@ import {AuthenticationService} from '../app/services/authentication/authenticati
 import {AuthenticationServiceMock} from '../app/services/mocks/authentication.service.mock';
 import {SubjectSetConstraint} from '../app/models/constraint-models/subject-set-constraint';
 
-describe('Integration test for query saving and restoring', () => {
+describe('Integration test for cohort saving and restoring', () => {
 
-  // mocked query objects
+  // mocked cohort objects
```
Comment (beaugogh) [inline src/tests/cohort.integration.spec.ts]: cohort definition no longer involves observation criteria and data table, thus removing them from tests

File: tslint.json [hunk 5]
```diff
@@ -85,7 +85,6 @@
         "variable-declaration": "nospace"
       }
     ],
-    "typeof-compare": true,
```
Comment (beaugogh) [inline tslint.json]: this is deprecated by ts 3

File: package.json [hunk 6]
```diff
@@ -45,15 +48,15 @@
     "zone.js": "^0.8.26"
   },
   "devDependencies": {
-    "@angular-devkit/build-angular": "~0.7.5",
-    "@angular/cli": "^6.1.5",
+    "@angular-devkit/build-angular": "~0.10.3",
+    "@angular/cli": "^7.0.3",
```
Comment (ewelinagr) [inline package.json]: We could update the version in [README](https://github.com/thehyve/glowing-bear#glowing-bear) too

Comment (beaugogh) [reply package.json]: updated :)

File: src/app/config/config.default.json [hunk 7]
```diff
@@ -6,11 +6,9 @@
   "doc-url": "https://glowingbear.app",
   "autosave-subject-sets": false,
   "show-observation-counts": true,
-  "instant-counts-update-1": false,
-  "instant-counts-update-2": false,
-  "instant-counts-update-3": false,
+  "instant-counts-update": false,
   "include-data-table": true,
-  "include-query-subscription": false,
+  "include-cohort-subscription": false,
```
Comment (ewelinagr) [inline src/app/config/config.default.json]: there are some inconsistencies in the naming now related to `query` -> `cohort` change, e.g.  here you renamed `include-query-subscription` to `include-cohort-subscription`, but there is still `DEFAULT_INCLUDE_QUERY_SUBSCRIPTION` in src/app/config/app.config.ts

File: src/app/modules/gb-cohort-selection-module/constraint-components/gb-constraint/gb-constraint.component.ts [hunk 8]
```diff
@@ -32,7 +32,7 @@ export class GbConstraintComponent implements OnInit {
               protected treeNodeService: TreeNodeService,
               protected resourceService: ResourceService,
               protected constraintService: ConstraintService,
-              protected queryService: QueryService,
+              protected queryService: CohortService,
```
Comment (ewelinagr) [inline src/app/modules/gb-cohort-selection-module/constraint-components/gb-constraint/gb-constraint.component.ts]: Here it could be also renamed: 
```suggestion
              protected cohortService: CohortService,
```

File: src/app/modules/gb-analysis-module/cross-table-components/gb-droppable-zone/gb-droppable-zone.component.ts [hunk 9]
```diff
@@ -7,13 +7,13 @@
...
-import {TreeNodeService} from '../../../services/tree-node.service';
-import {ConstraintService} from '../../../services/constraint.service';
-import {DropMode} from '../../../models/drop-mode';
-import {MessageHelper} from '../../../utilities/message-helper';
-import {AxisType} from '../../../models/table-models/axis-type';
+import {Constraint} from '../../../../models/constraint-models/constraint';
+import {CrossTableService} from '../../../../services/cross-table.service';
+import {TreeNodeService} from '../../../../services/tree-node.service';
+import {ConstraintService} from '../../../../services/constraint.service';
+import {DropMode} from '../../../../models/drop-mode';
```
Comment (ewelinagr) [inline src/app/modules/gb-analysis-module/cross-table-components/gb-droppable-zone/gb-droppable-zone.component.ts]: A bit less related to your PR, but I wonder what the [`DropMode`](https://github.com/thehyve/glowing-bear/blob/b55b2b13fcb48d7d3af3685735b5ba7232088549/src/app/models/drop-mode.ts) enum does, it looks like only the `DropMode.TreeNode` is used. Should it be updated to have `Cohort` and `Variable` modes instead of some old ones? Or removed?

File: src/app/modules/gb-cohort-selection-module/constraint-components/gb-constraint/gb-constraint.component.ts [hunk 10]
```diff
@@ -81,11 +81,10 @@ export class GbConstraintComponent implements OnInit {
   }
 
   update() {
-    this.queryService.step = Step.I;
```
Comment (ewelinagr) [inline src/app/modules/gb-cohort-selection-module/constraint-components/gb-constraint/gb-constraint.component.ts]: the `Step` Enum can be removed then

File: src/app/config/config.default.json (hunk 7 above)
Comment (ewelinagr) [reply src/app/config/config.default.json]: also this part of `src/app/config/app.config.ts` should be changed:
```
case 'include-query-subscription': {
          value = AppConfig.DEFAULT_INCLUDE_QUERY_SUBSCRIPTION; break;
        }
````

File: src/app/modules/gb-cohort-selection-module/gb-cohort-selection.component.spec.ts [hunk 11]
```diff
@@ -96,15 +93,16 @@ describe('GbDataSelectionComponent', () => {
   });
 
   it('should save query', () => {
```
Comment (ewelinagr) [inline src/app/modules/gb-cohort-selection-module/gb-cohort-selection.component.spec.ts]:
```suggestion
  it('should save cohort', () => {
```

File: src/app/modules/gb-main-module/gb-main.component.ts [hunk 12]
```diff
@@ -32,7 +32,7 @@ export class GbMainComponent implements OnInit {
               private transmartPackerHttpService: TransmartPackerHttpService,
               private treeNodeService: TreeNodeService,
               private constraintService: ConstraintService,
-              private queryService: QueryService) {
+              private queryService: CohortService) {
```
Comment (ewelinagr) [inline src/app/modules/gb-main-module/gb-main.component.ts]:
```suggestion
              private cohortService: CohortService) {
```

File: src/app/modules/gb-export-module/data-table-components/gb-data-table-dimensions/gb-data-table-dimensions.component.ts [hunk 13]
```diff
@@ -19,18 +19,16 @@ import {QueryService} from '../../../../services/query.service';
 export class GbDataTableDimensionsComponent implements OnInit {
 
   constructor(private dataTableService: DataTableService,
-              private queryService: QueryService) {
+              private queryService: CohortService) {
```
Comment (ewelinagr) [inline src/app/modules/gb-export-module/data-table-components/gb-data-table-dimensions/gb-data-table-dimensions.component.ts]:
```suggestion
              private cohortService: CohortService) {
```

File: src/app/services/navbar.service.ts [hunk 14]
```diff
@@ -26,20 +28,20 @@ export class NavbarService {
 
   constructor(private authService: AuthenticationService,
               private exportService: ExportService,
-              private queryService: QueryService) {
+              private queryService: CohortService) {
```
Comment (ewelinagr) [inline src/app/services/navbar.service.ts]:
```suggestion
              private cohortService: CohortService) {
```

File: src/app/modules/gb-export-module/data-table-components/gb-data-table-dimensions/gb-data-table-dimensions.component.ts (hunk 13 above)
Comment (ewelinagr) [reply src/app/modules/gb-export-module/data-table-components/gb-data-table-dimensions/gb-data-table-dimensions.component.ts]: although it seems not to be used, maybe the `CohortService` does not have to be injected here

File: src/app/services/resource.service.ts [hunk 15]
```diff
@@ -35,7 +35,9 @@ import {TransmartTrialVisit} from '../models/transmart-models/transmart-trial-vi
 import {CategoricalAggregate} from '../models/aggregate-models/categorical-aggregate';
 import {TransmartResourceService} from './transmart-services/transmart-resource.service';
 
-@Injectable()
+@Injectable({
+  providedIn: 'root',
```
Comment (ewelinagr) [inline src/app/services/resource.service.ts]: Why this change, if we still have ResourceService specified in providers of root module? Why only for this service?

File: src/app/services/navbar.service.ts [hunk 16]
```diff
@@ -26,20 +28,20 @@ export class NavbarService {
...
       if (exportEnabled) {
-        this.items.push({label: 'Export', routerLink: '/export'});
+        this.items.push({label: 'Data Table & Export', routerLink: '/export'});
       }
     });
   }
 
   updateNavbar(whichStep: string) {
-    this.isDataSelection = (whichStep === 'data-selection' || whichStep === '');
+    this.isDataSelection = (whichStep === 'cohort-selection' || whichStep === '');
```
Comment (ewelinagr) [inline src/app/services/navbar.service.ts]: navbar.service.mock also needs this change

**Comment by ewelinagr** [comment]
these changes will break lots of cypress tests, but they could be fixed later ([TMT-567](https://jira.thehyve.nl/browse/TMT-567))

File: src/app/config/config.default.json [hunk 17]
```diff
@@ -6,11 +6,9 @@
   "doc-url": "https://glowingbear.app",
   "autosave-subject-sets": false,
   "show-observation-counts": true,
-  "instant-counts-update-1": false,
-  "instant-counts-update-2": false,
-  "instant-counts-update-3": false,
```
Comment (ewelinagr) [inline src/app/config/config.default.json]: These properties are still in the README, it should be updated

File: src/app/services/mocks/constraint.service.mock.ts [hunk 18]
```diff
@@ -23,6 +23,8 @@ export class ConstraintServiceMock {
   conceptConstraints: Constraint[] = [];
   conceptLabels: string[] = [];
   allConstraints: Constraint[] = [];
+  variables: Concept[] = [];
+  variablesUpdated: Subject<Concept[]> = new Subject<Concept[]>();
```
Comment (ewelinagr) [inline src/app/services/mocks/constraint.service.mock.ts]: constraint_1() was removed from the service, but is still in the mock

File: src/app/services/resource.service.ts (hunk 15 above)
Comment (beaugogh) [reply src/app/services/resource.service.ts]: indeed, single services can be moved out of appModule
